src/qz/printer/status/CupsUtils.java

```diff
@@ -0,0 +1,240 @@
+package qz.printer.status;
+
+import com.sun.jna.Pointer;
+import org.codehaus.jettison.json.JSONArray;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import qz.utils.ShellUtilities;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Created by kyle on 5/17/17.
+ */
+public class CupsUtils {
+    private static final Logger log = LoggerFactory.getLogger(CupsUtils.class);
+
+    private static boolean httpInitialised = false;
+    private static Pointer http;
+    private static int cupsPort;
+    private static int subscriptionID = -1;
+
+
+    synchronized static void initCupsStuff() {
+        if (!httpInitialised) {
+            httpInitialised = true;
+            cupsPort = Cups.INSTANCE.ippPort();
+            http = Cups.INSTANCE.httpConnectEncrypt(
+                    Cups.INSTANCE.cupsServer(),
+                    Cups.INSTANCE.ippPort(),
+                    Cups.INSTANCE.cupsEncryption());
+        }
+    }
+
+    static Pointer listSubscriptions() {
+        Pointer request = Cups.INSTANCE.ippNewRequest(Cups.INSTANCE.ippOpValue("Get-Subscriptions"));
+
+        Cups.INSTANCE.ippAddString(request,
+                                   Cups.INSTANCE.ippTagValue("Operation"),
+                                   Cups.INSTANCE.ippTagValue("uri"),
+                                   "printer-uri",
+                                   "",
+                                   "ipp://localhost:" + cupsPort + "/printers/");
+        Cups.INSTANCE.ippAddString(request,
+                                   Cups.INSTANCE.ippTagValue("Operation"),
+                                   Cups.INSTANCE.ippTagValue("Name"),
+                                   "requesting-user-name",
+                                   "",
+                                   System.getProperty("user.name"));
+
+        return Cups.INSTANCE.cupsDoRequest(http, request, "/");
+    }
+
+    public static PrinterStatus[] getStatuses(String printerName) {
+        Pointer request = Cups.INSTANCE.ippNewRequest(Cups.INSTANCE.ippOpValue("Get-Printer-Attributes"));
+
+        Cups.INSTANCE.ippAddString(request,
+                                   Cups.INSTANCE.ippTagValue("Operation"),
+                                   Cups.INSTANCE.ippTagValue("uri"),
+                                   "printer-uri",
+                                   "",
+                                   //todo does this need to be sanitized?
```

Comment:
Cups doesn't allow spaces or special characters, but it shouldn't hurt to sanitize it anyway incase this behavior changes.